service_container.rst

```diff
@@ -842,6 +801,184 @@ need to make your service public, just override this setting:
+
+You've already seen that you can import many services at once by using the ``resource``
+key. For example, the default Symfony configuration contains this:
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # app/config/services.yml
+            services:
+                # ...
+
+                # the namespace prefix for classes (must end in \)
+                AppBundle\:
+                    # accepts a glob pattern
+                    resource: '../../src/AppBundle/*'
+                    # exclude some paths
+                    exclude: '../../src/AppBundle/{Entity,Repository}'
+
+                # these were imported above, but we want to add some extra config
+                AppBundle\Controller\:
+                    resource: '../../src/AppBundle/Controller'
+                    # apply some configuration to these services
+                    public: true
+                    tags: ['controller.service_arguments']
+
+        .. code-block:: xml
+
+            <!-- app/config/services.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <services>
+                    <!-- ... -->
+
+                    <prototype namespace="AppBundle\" resource="../../src/AppBundle/*" exclude="../../src/AppBundle/{Entity,Repository}" />
+
+                    <prototype namespace="AppBundle\Controller\" resource="../../src/AppBundle/Controller" public="true">
+                        <tag name="controller.service_arguments" />
+                    </prototype>
+                </services>
+            </container>
+
+This can be used to quickly make many classes available as services and apply some
+default configuration. The ``id`` of each service is its fully-qualified class name.
+You can override any service that's imported by using its id (class name) below
+(e.g. see :ref:`services-manually-wire-args`). If you override a service, none of
+the options (e.g. ``public``) are inherited from the import (but the overridden
+service *does* still inherit from ``_defaults``).
+
+.. note::
+
+    Wait, does this mean that *every* class in ``src/AppBundle`` is registered as
+    a service? Even model or entity classes? Actually, no. As long as you have
+    ``public: false`` under your ``_defaults`` key (or you can add it under the
+    specific import), all the imported services are *private*. Thanks to this, all
+    classes in ``src/AppBundle`` that are *not* explicitly used as services are
+    automatically removed from the final container. In reality, the import simply
+    means that all classes are "available to be *used* as services" without needing
+    to be manually configured.
+
```

Comment:
Maybe we could also note that value object classes with type hinted constructor should be explicitly excluded to avoid issues at compile time.